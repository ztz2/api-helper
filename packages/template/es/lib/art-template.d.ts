import artTemplate from './utils/template/art-template-web.js';
export default artTemplate;
