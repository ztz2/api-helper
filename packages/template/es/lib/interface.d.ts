import PrettierrcOptions from '@api-helper/cli/lib/tools/prettierrc-options';
export declare class DocumentConfig {
    id: string;
    title: string;
    url: string;
    type: string;
    auth: {
        username: string;
        password: string;
    };
    authToken: string;
    dataKey: string;
    /** 以下属性是基于项目级别的通用配置项 - 开始 * */
    exportFileId: string;
    exportFilePath: string;
    apiTplId: string;
    modelTplId: string;
    headCodeText: string;
    onlyApiFunc: boolean;
    maxlength: number;
    placeholder: boolean;
    grid: boolean;
    generateLabel: boolean;
    prettierrcOptions: PrettierrcOptions;
    /** 以下属性是基于项目级别的通用配置项 - 结束 * */
    constructor(id?: string | Partial<DocumentConfig>);
}
export declare class Template {
    label: string;
    value: string;
    content: string;
    builtIn: boolean;
    formatCodeExtension: "" | ".ts" | ".tsx" | ".js" | ".jsx" | ".vue" | ".html" | ".scss" | ".less" | ".css" | ".json" | ".java";
    constructor(options?: Partial<Template>);
}
export declare class FileDirectory {
    label: string;
    value: string;
    isFolder: boolean;
    apiId: string;
    modelTplId: string;
    modelTplIndex: number;
    templateContent: string;
    children: FileDirectory[];
    constructor(options?: Partial<FileDirectory>);
}
export declare class ExportFile {
    label: string;
    value: string;
    builtIn: boolean;
    formatCodeExtension: "" | ".ts" | ".tsx" | ".js" | ".jsx" | ".vue" | ".html" | ".scss" | ".less" | ".css" | ".json" | ".java";
    fileDirectory: FileDirectory[];
    exportFilePath: string;
    constructor(options?: Partial<FileDirectory>);
}
export declare function createTemplate(id?: string | Partial<DocumentConfig>): DocumentConfig;
export declare function createProjectConfig(id?: string | Partial<DocumentConfig>): DocumentConfig;
export declare function createFileDirectory(options?: Partial<FileDirectory>): FileDirectory;
export declare function createExportFile(options?: Partial<FileDirectory>): ExportFile;
export interface ChangeCase {
    /**
     * @example
     * changeCase.camelCase('test string') // => 'testString'
     */
    camelCase: (value: string) => string;
    /**
     * @example
     * changeCase.constantCase('test string') // => 'TEST_STRING'
     */
    constantCase: (value: string) => string;
    /**
     * @example
     * changeCase.dotCase('test string') // => 'test.string'
     */
    dotCase: (value: string) => string;
    /**
     * @example
     * changeCase.headerCase('test string') // => 'Test-String'
     */
    headerCase: (value: string) => string;
    /**
     * @example
     * changeCase.lowerCase('TEST STRING') // => 'test string'
     */
    lowerCase: (value: string) => string;
    /**
     * @example
     * changeCase.lowerCaseFirst('TEST') // => 'tEST'
     */
    lowerCaseFirst: (value: string) => string;
    /**
     * @example
     * changeCase.paramCase('test string') // => 'test-string'
     */
    paramCase: (value: string) => string;
    /**
     * @example
     * changeCase.pascalCase('test string') // => 'TestString'
     */
    pascalCase: (value: string) => string;
    /**
     * @example
     * changeCase.pathCase('test string') // => 'test/string'
     */
    pathCase: (value: string) => string;
    /**
     * @example
     * changeCase.sentenceCase('testString') // => 'Test string'
     */
    sentenceCase: (value: string) => string;
    /**
     * @example
     * changeCase.snakeCase('test string') // => 'test_string'
     */
    snakeCase: (value: string) => string;
    /**
     * @example
     * changeCase.swapCase('Test String') // => 'tEST sTRING'
     */
    swapCase: (value: string) => string;
    /**
     * @example
     * changeCase.titleCase('a simple test') // => 'A Simple Test'
     */
    titleCase: (value: string) => string;
    /**
     * @example
     * changeCase.upperCase('test string') // => 'TEST STRING'
     */
    upperCase: (value: string) => string;
    /**
     * @example
     * changeCase.upperCaseFirst('test') // => 'Test'
     */
    upperCaseFirst: (value: string) => string;
}
