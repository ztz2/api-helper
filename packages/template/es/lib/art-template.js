var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
// @ts-ignore
import artTemplate from '../lib/utils/template/art-template-web.js';
// @ts-ignore
artTemplate.defaults.escape = false;
artTemplate.defaults.minimize = false;
// 新增模板《》语法的界定符规则
artTemplate.defaults.rules[2] = __assign(__assign({}, artTemplate.defaults.rules[1]), { test: /《([@#]?)[ \t]*(\/?)([\w\W]*?)[ \t]*》/ });
artTemplate.defaults.imports.renderCharacterByBoolean = function (booleanStatus, trueValue, falseValue) {
    if (booleanStatus === void 0) { booleanStatus = true; }
    if (trueValue === void 0) { trueValue = ''; }
    if (falseValue === void 0) { falseValue = ''; }
    return booleanStatus ? trueValue : falseValue;
};
artTemplate.defaults.imports.renderTemplateLiteral = function (booleanStatus, defaultValue) {
    if (booleanStatus === void 0) { booleanStatus = true; }
    if (defaultValue === void 0) { defaultValue = '\''; }
    return booleanStatus ? '`' : defaultValue;
};
export default artTemplate;
