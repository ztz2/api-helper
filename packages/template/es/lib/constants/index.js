export var API_CUSTOM_TEMPLATE_ID = 'API自定义分组ID_ea18930468d9402e';
export var MODEL_CUSTOM_TEMPLATE_ID = '模型自定义分组ID_6f3f32d6b404aaf9';
export var DEFAULT_SELECT_API_TPL_ID = 'typescript_api_76ea205dba9f3dfb';
export var DEFAULT_SELECT_MODEL_TPL_ID = 'javascript_map_841d8ffc9b34d1d5';
