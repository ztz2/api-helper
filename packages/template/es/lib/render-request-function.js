import * as changeCase from 'change-case';
import { getSchema } from '@api-helper/core/lib/helpers';
import artTemplate from '../lib/art-template';
import formatCode from '../lib/utils/prettier';
import { renderInterfaceName } from '../lib/render-interface';
export function renderRequestFunction(api, options) {
    if (!api) {
        return '';
    }
    var codeType = (options === null || options === void 0 ? void 0 : options.codeType) || 'typescript';
    var dataKey = options === null || options === void 0 ? void 0 : options.dataKey;
    var onRenderRequestFunctionName = (options && options.onRenderRequestFunctionName) ? options.onRenderRequestFunctionName : renderRequestFunctionName;
    var onRenderInterfaceName = (options && options.onRenderInterfaceName) ? options.onRenderInterfaceName : renderInterfaceName;
    var responseDataSchema = dataKey ? getSchema(api.responseDataSchema, dataKey) : api.responseDataSchema;
    var templateTenderParams = {
        api: api,
        isTypescript: codeType === 'typescript',
        commentCode: renderRequestFunctionComment(api),
        formDataKeyNameListStr: JSON.stringify(api.formDataKeyNameList),
        pathParamKeyNameListStr: JSON.stringify(api.pathParamKeyNameList),
        queryStringKeyNameListStr: JSON.stringify(api.queryStringKeyNameList),
        requestFunctionName: onRenderRequestFunctionName(api, { changeCase: changeCase }),
        requestDataInterfaceName: onRenderInterfaceName(api.requestDataSchema, api, { paramType: 'request', changeCase: changeCase }),
        requestExtraDataInterfaceName: onRenderInterfaceName(api.requestExtraDataSchema, api, { paramType: 'request', isExtraData: true, changeCase: changeCase }),
        responseDataInterfaceName: onRenderInterfaceName(responseDataSchema, api, { paramType: 'response', changeCase: changeCase }),
    };
    var code = artTemplate.render("{{if commentCode}}{{commentCode}}\n{{/if}}export function {{requestFunctionName}}(data{{if isTypescript}}: {{requestDataInterfaceName}}{{/if}}, extraData{{if isTypescript}}?: {{if api.requestExtraDataSchema}}{{requestExtraDataInterfaceName}}{{else}}unknown{{/if}}{{/if}}, ...args{{if isTypescript}}: CurrentRequestFunctionRestArgsType{{/if}}) {\n  return request{{if isTypescript}}<{{responseDataInterfaceName}}>{{/if}}(\n    processRequestFunctionConfig(data, extraData, {{requestFunctionName}}.requestConfig),\n    ...args\n  );\n}\n{{requestFunctionName}}.requestConfig = {\n  path: '{{api.path}}',\n  method: '{{api.method}}',\n  formDataKeyNameList: {{formDataKeyNameListStr}},\n  pathParamKeyNameList: {{pathParamKeyNameListStr}},\n  queryStringKeyNameList: {{queryStringKeyNameListStr}}\n}", templateTenderParams);
    return formatCode(code, {
        parser: codeType !== 'typescript' ? 'babel' : 'typescript'
    });
}
export function renderRequestFunctionName(api, options) {
    return changeCase.camelCase("".concat(api.path, " By ").concat(api.method));
}
function renderRequestFunctionComment(api) {
    var templateTenderParams = {
        api: api,
        apiDescription: [api.summary, api.description].filter(Boolean).join('„ÄÅ')
    };
    return artTemplate.render("/**\n   * @description{{if apiDescription}} {{apiDescription}}{{else}} \u65E0{{/if}}\n{{if api.docURL}}   * @doc {{api.docURL}}\n{{/if}}   * @url [ {{api.method.toUpperCase()}} ] {{api.path}}\n   */", templateTenderParams);
}
