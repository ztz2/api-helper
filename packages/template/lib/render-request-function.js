"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.renderRequestFunctionComment = exports.renderRequestFunctionName = exports.renderRequestFunction = void 0;
var _changeCase = __importStar(require("change-case"));
var helpers_1 = require("@api-helper/core/lib/helpers");
var art_template_1 = __importDefault(require("../lib/art-template"));
var render_interface_1 = require("../lib/render-interface");
var util_1 = require("../lib/utils/util");
function renderRequestFunction(api, options) {
    var _a;
    if (!api) {
        return '';
    }
    var codeType = (options === null || options === void 0 ? void 0 : options.codeType) || 'typescript';
    var dataKey = options === null || options === void 0 ? void 0 : options.dataKey;
    var isEmptyRequestData = (0, util_1.isEmptySchema)(api.requestDataSchema);
    var emptyRequestDataValue = isEmptyRequestData ? ((_a = api.requestDataSchema) === null || _a === void 0 ? void 0 : _a.type) === 'array' ? '[]' : '{}' : '';
    var onRenderRequestFunctionName = (options && options.onRenderRequestFunctionName) ? options.onRenderRequestFunctionName : renderRequestFunctionName;
    var onRenderInterfaceName = (options && options.onRenderInterfaceName) ? options.onRenderInterfaceName : render_interface_1.renderInterfaceName;
    var responseDataSchema = dataKey ? (0, helpers_1.getSchema)(api.responseDataSchema, dataKey) : api.responseDataSchema;
    var requestFunctionName = onRenderRequestFunctionName(api, {
        changeCase: _changeCase,
    });
    var requestDataInterfaceName = onRenderInterfaceName(api, {
        paramType: 'request',
        changeCase: _changeCase,
        schema: api.requestDataSchema,
    });
    var requestExtraDataInterfaceName = onRenderInterfaceName(api, {
        isExtraData: true,
        paramType: 'request',
        changeCase: _changeCase,
        schema: api.requestExtraDataSchema,
    });
    var responseDataInterfaceName = onRenderInterfaceName(api, {
        paramType: 'response',
        changeCase: _changeCase,
        schema: responseDataSchema,
    });
    var templateTenderParams = {
        api: api,
        emptyRequestDataValue: emptyRequestDataValue,
        isTypescript: codeType === 'typescript',
        commentCode: renderRequestFunctionComment(api),
        formDataKeyNameListStr: JSON.stringify(api.formDataKeyNameList),
        pathParamKeyNameListStr: JSON.stringify(api.pathParamKeyNameList),
        queryStringKeyNameListStr: JSON.stringify(api.queryStringKeyNameList),
        requestFunctionName: requestFunctionName,
        requestDataInterfaceName: requestDataInterfaceName,
        requestExtraDataInterfaceName: requestExtraDataInterfaceName,
        responseDataInterfaceName: responseDataInterfaceName,
    };
    var code = art_template_1.default.render("\u300Aif commentCode\u300B\u300AcommentCode\u300B\n\u300A/if\u300Bexport function \u300ArequestFunctionName\u300B(data\u300Aif isTypescript\u300B: \u300ArequestDataInterfaceName\u300B\u300A/if\u300B\u300Aif emptyRequestDataValue\u300B = \u300AemptyRequestDataValue\u300B\u300A/if\u300B, extraData\u300Aif isTypescript\u300B?: \u300Aif api.requestExtraDataSchema\u300B\u300ArequestExtraDataInterfaceName\u300B\u300Aelse\u300Bunknown\u300A/if\u300B\u300A/if\u300B, ...args\u300Aif isTypescript\u300B: CurrentRequestFunctionRestArgsType\u300A/if\u300B) {\n  return request\u300Aif isTypescript\u300B<\u300AresponseDataInterfaceName\u300B>\u300A/if\u300B(\n    processRequestFunctionConfig(data, extraData, \u300ArequestFunctionName\u300B.requestConfig),\n    ...args\n  );\n}\n\u300ArequestFunctionName\u300B.requestConfig = {\n  path: '\u300Aapi.path\u300B',\n  method: '\u300Aapi.method.toUpperCase()\u300B',\n  formDataKeyNameList: \u300AformDataKeyNameListStr\u300B,\n  pathParamKeyNameList: \u300ApathParamKeyNameListStr\u300B,\n  queryStringKeyNameList: \u300AqueryStringKeyNameListStr\u300B\n}", templateTenderParams);
    return code;
}
exports.renderRequestFunction = renderRequestFunction;
function renderRequestFunctionName(api, options) {
    var _a;
    var changeCase = (_a = options === null || options === void 0 ? void 0 : options.changeCase) !== null && _a !== void 0 ? _a : _changeCase;
    return changeCase.camelCase("".concat(api.path, " By ").concat(api.method));
}
exports.renderRequestFunctionName = renderRequestFunctionName;
function renderRequestFunctionComment(api, updateTime) {
    if (updateTime === void 0) { updateTime = ''; }
    return art_template_1.default.render("/**\n   * @description\u300Aif description\u300B \u300Adescription\u300B\u300Aelse\u300B \u65E0\u300A/if\u300B\u300Aif docURL\u300B\n   * @doc \u300AdocURL\u300B\u300A/if\u300B\n   * @url \u300Aurl\u300B\u300Aif updateTime\u300B\n   * @update \u300AupdateTime\u300B\u300A/if\u300B\n   */", {
        description: [api.title, api.description].filter(Boolean).join('„ÄÅ'),
        docURL: api.docURL,
        url: "[ ".concat(api.method.toUpperCase(), " ] ").concat(api.path),
        updateTime: updateTime
    });
}
exports.renderRequestFunctionComment = renderRequestFunctionComment;
