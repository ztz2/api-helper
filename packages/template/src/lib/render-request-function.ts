import * as _changeCase from 'change-case';
import { APIHelper } from '@api-helper/core/lib/types';
import { getSchema } from '@api-helper/core/lib/helpers';

import { ChangeCase } from '@/lib/types';
import artTemplate from '@/lib/art-template';
import { renderInterfaceName } from '@/lib/render-interface';
import { isEmptySchema } from '@/lib/utils/util';

export function renderRequestFunction(
  api: APIHelper.API,
  options: {
    codeType?: 'typescript' | 'javascript';
    dataKey?: string | undefined;
    genHeaders?: boolean;
    genCookies?: boolean;
    genResponseContentType?: boolean;
    onRenderInterfaceName?: typeof renderInterfaceName;
    onRenderRequestFunctionName?: typeof renderRequestFunctionName;
  } = {}
) {
  if (!api) {
    return '';
  }
  const dataKey = options.dataKey;
  const codeType = options.codeType || 'typescript';
  const isEmptyRequestData = isEmptySchema(api.requestDataSchema);
  const emptyRequestDataValue = isEmptyRequestData ? api.requestDataSchema?.type === 'array' ? '[]' : '{}' : '';
  const onRenderRequestFunctionName = options.onRenderRequestFunctionName ?? renderRequestFunctionName;
  const onRenderInterfaceName = options.onRenderInterfaceName ?? renderInterfaceName;
  const responseDataSchema = dataKey ? getSchema(api.responseDataSchema, dataKey) : api.responseDataSchema;

  const requestFunctionName = onRenderRequestFunctionName(api, {
    changeCase: _changeCase,
  });
  const requestDataInterfaceName = onRenderInterfaceName(api, {
    paramType: 'request',
    changeCase: _changeCase,
    schema: api.requestDataSchema as APIHelper.Schema,
  });
  const requestExtraDataInterfaceName = onRenderInterfaceName(api, {
    isExtraData: true,
    paramType: 'request',
    changeCase: _changeCase,
    schema: api.requestExtraDataSchema as APIHelper.Schema,
  });
  const responseDataInterfaceName = onRenderInterfaceName(api, {
    paramType: 'response',
    changeCase: _changeCase,
    schema: responseDataSchema,
  });

  const headers = api.headers?.params?.map?.((item) => item.keyName)?.filter?.(Boolean) ?? [];
  const cookies = api.cookies?.params?.map?.((item) =>  item.keyName)?.filter?.(Boolean) ?? [];

  const templateTenderParams = {
    api,
    options,
    emptyRequestDataValue,
    requestFunctionName,
    requestDataInterfaceName,
    requestExtraDataInterfaceName,
    responseDataInterfaceName,
    isTypescript: codeType === 'typescript',
    commentCode: renderRequestFunctionComment(api),
    formDataKeyNameListStr: JSON.stringify(api.formDataKeyNameList),
    pathParamKeyNameListStr: JSON.stringify(api.pathParamKeyNameList),
    queryStringKeyNameListStr: JSON.stringify(api.queryStringKeyNameList),
    headersStr: JSON.stringify(headers),
    cookiesStr: JSON.stringify(cookies),
    requestContentTypeStr: JSON.stringify(api.requestContentType ?? []),
    responseContentTypeStr: JSON.stringify(api.responseContentType ?? []),
  };
  const code = artTemplate.render(
    `《if commentCode》《commentCode》
《/if》export function 《requestFunctionName》(data《if isTypescript》: 《requestDataInterfaceName》《/if》《if emptyRequestDataValue》 = 《emptyRequestDataValue》《/if》, extraData《if isTypescript》?: 《if api.requestExtraDataSchema》《requestExtraDataInterfaceName》《else》unknown《/if》《/if》, ...args《if isTypescript》: CurrentRequestFunctionRestArgsType《/if》) {
  return request《if isTypescript》<《responseDataInterfaceName》>《/if》(
    processRequestFunctionConfig(data, extraData, 《requestFunctionName》.requestConfig),
    ...args
  );
}
《requestFunctionName》.requestConfig = {
  path: '《api.path》',
  method: '《api.method.toUpperCase()》',《if options.genHeaders》
  headers: 《headersStr》,《/if》《if options.genCookies》
  cookies: 《cookiesStr》,《/if》
  requestContentType: 《requestContentTypeStr》,《if options.genResponseContentType》
  responseContentType: 《responseContentTypeStr》,《/if》
  formDataKeyNameList: 《formDataKeyNameListStr》,
  pathParamKeyNameList: 《pathParamKeyNameListStr》,
  queryStringKeyNameList: 《queryStringKeyNameListStr》,
}`, templateTenderParams);

  return code;
}

export function renderRequestFunctionName(
  api: APIHelper.API,
  options: { changeCase: ChangeCase }) {
  const changeCase = options?.changeCase ?? _changeCase;
  return changeCase.camelCase(`${api.path} By ${api.method}`);
}

export function renderRequestFunctionComment(api: APIHelper.API, updateTime = '') {
  return artTemplate.render(
`/**
   * @description《if description》 《description》《else》 无《/if》《if docURL》
   * @doc 《docURL》《/if》
   * @url 《url》《if updateTime》
   * @update 《updateTime》《/if》
   */`, {
      description: [api.title, api.description].filter(Boolean).join('、'),
      docURL: api.docURL,
      url: `[ ${api.method.toUpperCase()} ] ${api.path}`,
      updateTime
    });
}
