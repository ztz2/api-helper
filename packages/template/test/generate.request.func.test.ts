import { join } from 'path';
import ParserSwagger from '@api-helper/core/es/lib/parser/parser-swagger';

import { readJsonFile } from './utils/file';
import { renderRequestFunction } from '../src/lib/render-request-function';

describe('生成接口函数测试', () => {
  test('OpenAPI-2.0生成接口函数测试', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-2.0.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const codeBox: string[] = [];
    for (const api of category.apiList) {
      codeBox.push(renderRequestFunction(api));
    }
    expect(
      codeBox.join('\n'),
    ).toMatchSnapshot('OpenAPI-2.0生成接口函数测试');
  });

  test('OpenAPI-3.0生成接口函数测试', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-3.0.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const codeBox: string[] = [];
    for (const api of category.apiList) {
      codeBox.push(renderRequestFunction(api));
    }
    expect(
      codeBox.join('\n'),
    ).toMatchSnapshot('OpenAPI-3.0生成接口函数测试');
  });
});
