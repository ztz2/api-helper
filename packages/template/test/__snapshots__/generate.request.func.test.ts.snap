// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`生成接口函数测试 OpenAPI-2.0生成接口函数测试: OpenAPI-2.0生成接口函数测试 1`] = `
"/**
 * @描述 Place an order for a pet
 * @请求信息 [ POST ] /v2/store/order
 */
export function v2StoreOrderByPost(
  data: IV2StoreOrderRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<IV2StoreOrderResponseByPost>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreOrderByPost.requestConfig
    ),
    ...args
  );
}
v2StoreOrderByPost.requestConfig = {
  path: '/v2/store/order',
  method: 'post',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: []
};

/**
 * @描述 Find purchase order by ID、For valid response try integer IDs with value >= 1 and <= 10. Other values will generated exceptions
 * @请求信息 [ GET ] /v2/store/order/{orderId}
 */
export function v2StoreOrderOrderIdByGet(
  data: IV2StoreOrderOrderIdRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<IV2StoreOrderOrderIdResponseByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreOrderOrderIdByGet.requestConfig
    ),
    ...args
  );
}
v2StoreOrderOrderIdByGet.requestConfig = {
  path: '/v2/store/order/{orderId}',
  method: 'get',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['orderId'],
  queryStringKeyNameList: []
};

/**
 * @描述 Delete purchase order by ID、For valid response try integer IDs with positive integer value. Negative or non-integer values will generate API errors
 * @请求信息 [ DELETE ] /v2/store/order/{orderId}
 */
export function v2StoreOrderOrderIdByDelete(
  data: IV2StoreOrderOrderIdRequestByDelete,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2StoreOrderOrderIdResponseTypeByDelete>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreOrderOrderIdByDelete.requestConfig
    ),
    ...args
  );
}
v2StoreOrderOrderIdByDelete.requestConfig = {
  path: '/v2/store/order/{orderId}',
  method: 'delete',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['orderId'],
  queryStringKeyNameList: []
};

/**
 * @描述 Returns pet inventories by status、Returns a map of status codes to quantities
 * @请求信息 [ GET ] /v2/store/inventory
 */
export function v2StoreInventoryByGet(
  data: V2StoreInventoryRequestTypeByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2StoreInventoryResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreInventoryByGet.requestConfig
    ),
    ...args
  );
}
v2StoreInventoryByGet.requestConfig = {
  path: '/v2/store/inventory',
  method: 'get',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: []
};
"
`;

exports[`生成接口函数测试 OpenAPI-3.0生成接口函数测试: OpenAPI-3.0生成接口函数测试 1`] = `
"/**
 * @描述 Returns pet inventories by status、Returns a map of status codes to quantities
 * @请求信息 [ GET ] /api/v3/store/inventory
 */
export function apiV3StoreInventoryByGet(
  data: ApiV3StoreInventoryRequestTypeByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<ApiV3StoreInventoryResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      apiV3StoreInventoryByGet.requestConfig
    ),
    ...args
  );
}
apiV3StoreInventoryByGet.requestConfig = {
  path: '/api/v3/store/inventory',
  method: 'get',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: []
};

/**
 * @描述 Place an order for a pet、Place a new order in the store
 * @请求信息 [ POST ] /api/v3/store/order
 */
export function apiV3StoreOrderByPost(
  data: IApiV3StoreOrderRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<ApiV3StoreOrderResponseTypeByPost>(
    processRequestFunctionConfig(
      data,
      extraData,
      apiV3StoreOrderByPost.requestConfig
    ),
    ...args
  );
}
apiV3StoreOrderByPost.requestConfig = {
  path: '/api/v3/store/order',
  method: 'post',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [
    'id',
    'petId',
    'quantity',
    'shipDate',
    'status',
    'complete'
  ]
};

/**
 * @描述 Find purchase order by ID、For valid response try integer IDs with value <= 5 or > 10. Other values will generate exceptions.
 * @请求信息 [ GET ] /api/v3/store/order/{orderId}
 */
export function apiV3StoreOrderOrderIdByGet(
  data: IApiV3StoreOrderOrderIdRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<ApiV3StoreOrderOrderIdResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      apiV3StoreOrderOrderIdByGet.requestConfig
    ),
    ...args
  );
}
apiV3StoreOrderOrderIdByGet.requestConfig = {
  path: '/api/v3/store/order/{orderId}',
  method: 'get',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['orderId'],
  queryStringKeyNameList: []
};

/**
 * @描述 Delete purchase order by ID、For valid response try integer IDs with value < 1000. Anything above 1000 or nonintegers will generate API errors
 * @请求信息 [ DELETE ] /api/v3/store/order/{orderId}
 */
export function apiV3StoreOrderOrderIdByDelete(
  data: IApiV3StoreOrderOrderIdRequestByDelete,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<ApiV3StoreOrderOrderIdResponseTypeByDelete>(
    processRequestFunctionConfig(
      data,
      extraData,
      apiV3StoreOrderOrderIdByDelete.requestConfig
    ),
    ...args
  );
}
apiV3StoreOrderOrderIdByDelete.requestConfig = {
  path: '/api/v3/store/order/{orderId}',
  method: 'delete',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['orderId'],
  queryStringKeyNameList: []
};
"
`;
